interface/src/app/greenelab/utils.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * This module includes a few utility components that will be shared in other
+ * modules to avoid duplicate codes.
+ */
+
+angular.module('greenelab.utils', [])
+.constant('ApiBasePath', '/api/v0/')
+
+// A service function that returns a string of error message based on input
+// context and response.
+.value('errGen', function(context, response) {
```

Comment:
`value()` is a shortcut version of `factory()` method. If a service doesn't have any dependencies, then you can define the value to be any string, object or function. In this case, `errGen` is a function to compile error message based on input parameters.